indico/modules/events/registration/util.py

```diff
@@ -54,16 +56,36 @@ def user_registered_in_event(user, event):
                 .count())
 
 
-def get_event_section_data(regform, management=False):
-    return [s.view_data for s in regform.sections if not s.is_deleted and (management or not s.is_manager_only)]
+def get_event_section_data(regform, management=False, registration=None):
+    data = []
+    if not registration:
+        return [s.view_data for s in regform.sections if not s.is_deleted and (management or not s.is_manager_only)]
+    else:
+        registration_data = {r.field_data.field.id: r for r in registration.data}
+        for section in regform.sections:
+            if not section.is_deleted and (management or not section.is_manager_only):
+                section_data = section.own_data
+                section_data['items'] = []
+
+                for child in section.children:
+                    if not child.is_deleted:
+                        if isinstance(child.field_impl, ChoiceBaseField):
+                            field_data = get_field_merged_options(child, registration_data)
+                        else:
+                            field_data = child.view_data
+                        section_data['items'].append(field_data)
+                data.append(section_data)
+        return data
 
 
-def make_registration_form(regform, management=False):
+def make_registration_form(regform, management=False, registration=None):
     """Creates a WTForm based on registration form fields"""
 
     class RegistrationFormWTF(IndicoForm):
         def validate_email(self, field):
-            if regform.get_registration(email=field.data):
+            existing_registration = regform.get_registration(email=field.data)
+            if ((registration is None and existing_registration) or
+               (registration is not None and registration != existing_registration)):
```

Comment:
wrong indentation (if condition continuation uses 8 spaces instead of 4 spaces (or 3 like you have here ;)))